### ISSUE TITLE
BasicAuthenticator does not add the Authenticator Header to the replicator

### ISSUE THREAD

**Issue by cjhobbs** [couchbase/couchbase-lite-java-ce-root#7]
adding BasicAuthenticator to your project to restrict your users to channels doesn't work. I discovered with my team that the appropriate authentication headers are not being sent with the BasicAuthenticator namely Authenticator  = Basic YourBasicAuthenticatorNumber. Was able to work around this issue by adding the Authenticator as a custom header. This made all users log into the application as non verified users. Was only able to pull public documents.

**Comment by bmeike** [comment]
Can we get a little more information on this?  What version of CBL are you using?  Can you get the logs, preferably from both the SGW and the client side, for a login attempt

**Comment by cjhobbs** [comment]
I probably can't get you fresh logs right now. Because we fixed the issue by adding custom headers. If you wan't the logs I will direct you to [this](https://forums.couchbase.com/t/not-understanding-how-to-get-couchbase-lite-to-pull-down-all-exsisting-dbs-from-sync-gateway/25349/44) forum post. the code that I used to fix the issue is also at the bottom of the post. I could possibly try to run it again. But to be honest when reading the logs there wasn't much there to indicate that it was failing it just defaulted to the guest login and pulled only the public documents.
